config/src/config.rs

```diff
@@ -260,6 +260,15 @@ pub enum RoleType {
     FullNode,
 }
 
+impl RoleType {
+    pub fn get_role_string(&self) -> String {
```

Comment:
I don't think you actually need this method
You can just use `config.base.role`

Reply:
Don't I need it in swarmconfig::new because "template" doesn't have the right role?